dev/tests/js/jasmine/tests/app/code/Magento/Customer/frontend/js/customer-data.test.js

```diff
@@ -95,84 +97,100 @@ define([
     }
 
     describe('Magento_Customer/js/customer-data', function () {
+
         beforeAll(function () {
             clearLocalStorage();
         });
 
-        beforeEach(function () {
-            jQueryGetJSON = $.getJSON;
+        beforeEach(function (done) {
+            originalGetJSON = jQuery.getJSON;
```

Comment:
Seems like we have `$` in the top, so probably we should use it instead
```suggestion
            originalGetJSON = $.getJSON;
```

Reply:
Breaks the following tests:

`"getExpiredSectionNames method` -> `Check that result does not contain unexpired section names...` with error:
```
log: JQMIGRATE: jQuery.parseJSON requires a valid JSON string

log:
>> Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

`"getExpiredSectionNames method` -> `Check that result does not contain unsupported section names...` with error:
```
>> Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

`"reload" method` -> `Check it returns proper sections object when passed array with a single section name...` with error:
```
>> Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

`"reload" method` -> `Check it returns proper sections object when passed array with a single section name...` with error:
```
Expected Object({ readyState: 1, getResponseHeader: Function, getAllResponseHeaders: Function, setRequestHeader: Function, overrideMimeType: Function, statusCode: Function, abort: Function, state: Function, always: Function, then: Function, promise: Function, pipe: Function, done: Function, fail: Function, progress: Function, complete: Function, success: Function, error: Function }) to equal <jasmine.objectContaining(Object({ responseJSON: Object({ section: Object({  }) }) }))>. (1)
```

`"reload" method` -> `Check it returns proper sections object when passed array with multiple section names...`
```
Expected Object({ readyState: 1, getResponseHeader: Function, getAllResponseHeaders: Function, setRequestHeader: Function, overrideMimeType: Function, statusCode: Function, abort: Function, state: Function, always: Function, then: Function, promise: Function, pipe: Function, done: Function, fail: Function, progress: Function, complete: Function, success: Function, error: Function }) to equal <jasmine.objectContaining(Object({ responseJSON: Object({ cart: Object({  }), customer: Object({  }), messages: Object({  }) }) }))>. (1)
```

`"reload" method` -> `Check it returns all sections when passed wildcard string...`
```
Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

`"reload" method` -> `Check it returns all sections when passed wildcard string`
```
Expected spy unknown to have been called. (1)
       Expected Object({ readyState: 1, getResponseHeader: Function, getAllResponseHeaders: Function, setRequestHeader: Function, overrideMimeType: Function, statusCode: Function, abort: Function, state: Function, always: Function, then: Function, promise: Function, pipe: Function, done: Function, fail: Function, progress: Function, complete: Function, success: Function, error: Function }) to equal <jasmine.objectContaining(Object({ responseJSON: Object({ cart: Object({  }), customer: Object({  }), messages: Object({  }) }) }))>. (2)
```

`"invalidate" method` -> `Should be defined`
```
Error: [object Object] at
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/Magento_Customer/js/customer-data.js:87
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3238 fire
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:3368 fireWith
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:9848 done
>> http:/localhost:8000/pub/static/frontend/Magento/luma/en_US/jquery.js:10317 callback
```

Reply:
Figured out a way around this, and updated below.